### PR TITLE
feat(api): Add `Incident` data model, and endpoint to return a list of incidents (SEN-519, SEN-518)

### PR DESCRIPTION
**Pull request by wedamija**
Initial data model for `Incident`, will add more models as needed.

Trying a pattern where we add most incident related code to it's own django app rather than in
endpoints. Potentially endpoints could also live in this app rather than all in `api`, but not
going to make that change now. I can move the model just back into the models folder if anyone feels
strongly that this is the wrong pattern to use, but I feel like ideally we should be splitting
things up into more distinct folders, rather than one giant one.

### CODE REVIEW

File: src/sentry/api/serializers/models/incident.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        results = {}
+        for item in item_list:
+            results[item] = {'projects': incident_projects.get(item.id, [])}
+        return results
+
+    def serialize(self, obj, attrs, user):
+        return {
+            'id': six.text_type(obj.id),
+            'identifier': obj.identifier,
```
Comment (markstory) [inline src/sentry/api/serializers/models/incident.py]: Is this a URL safe slug?

File: src/sentry/testutils/factories.py [hunk 2]
```diff
@@ -834,3 +838,34 @@ def create_platform_external_issue(group=None, service_type=None,
...
+
+    @staticmethod
+    def create_incident(
+        organization, projects, identifier=None, detection_uuid=None, status=0,
+        title=None, query='test query', date_started=None, date_detected=None,
+        date_closed=None,
+    ):
+        # TODO: Remove this once we handle creating this as part of the object
+        if identifier is None:
+            identifier = Incident.objects.filter(organization=organization).count() + 1
```
Comment (markstory) [inline src/sentry/testutils/factories.py]: Could you run into concurrency races here?

Comment (wedamija) [reply src/sentry/testutils/factories.py]: This is just tests, so should be fine? I'm going to write this properly later on when we handle actually creating the incidents.

File: src/sentry/api/serializers/models/incident.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/api/serializers/models/incident.py]: Just an int. We couldn't think of a great name, but it represents the number you see here: https://www.figma.com/proto/I7oNk99EkE4GHbwGChCDUsJw/Incidents?node-id=322%3A1&scaling=min-zoom&redirected=1 (Incidents > #2381)

File: src/sentry/api/endpoints/organization_incident_index.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.incidents.models import Incident
+
+
+class OrganizationIncidentPermission(OrganizationPermission):
+    scope_map = {
+        'GET': ['org:read', 'org:write', 'org:admin'],
+        'POST': ['org:read', 'org:write', 'org:admin'],
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_incident_index.py]: Should people with only read access to the org be able to create incidents?

Comment (wedamija) [reply src/sentry/api/endpoints/organization_incident_index.py]: Oops, left this in from debugging CI

File: src/sentry/south_migrations/0476_incidents.py [hunk 4]
```diff
@@ -0,0 +1,1409 @@
...
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = True
+
+    def forwards(self, orm):
+        # Adding model 'Incident'
+        db.create_table('sentry_incident', (
+            ('id', self.gf('sentry.db.models.fields.bounded.BoundedBigAutoField')(primary_key=True)),
+            ('organization', self.gf('sentry.db.models.fields.foreignkey.FlexibleForeignKey')(
```
Comment (wedamija) [inline src/sentry/south_migrations/0476_incidents.py]: @mattrobenolt Could you bless this? I'm adding a foreign key to organization, which has worked ok before.

SQL:

```
 CREATE TABLE "sentry_incident" (
   "id" bigserial NOT NULL PRIMARY KEY,
   "organization_id" bigint NOT NULL,
   "identifier" integer NOT NULL,
   "detection_uuid" uuid NOT NULL,
   "status" smallint CHECK ("status" >= 0) NOT NULL,
   "title" text NOT NULL, "query" text NOT NULL,
   "date_started" timestamp with time zone NOT NULL,
   "date_detected" timestamp with time zone NOT NULL,
   "date_added" timestamp with time zone NOT NULL,
   "date_closed" timestamp with time zone NOT NULL
)
ALTER TABLE "sentry_incident"
  ADD CONSTRAINT "sentry_incident_organization_id_56facaa712033034_uniq" UNIQUE ("organization_id", "identifier")
CREATE TABLE "sentry_incidentproject" (
  "id" bigserial NOT NULL PRIMARY KEY,
  "project_id" integer CHECK ("project_id" >= 0) NOT NULL,
  "incident_id" bigint NOT NULL
)
ALTER TABLE "sentry_incidentproject"
  ADD CONSTRAINT "sentry_incidentproject_project_id_6bdb72a1d37644fb_uniq" UNIQUE ("project_id", "incident_id")
ALTER TABLE "sentry_incident"
  ADD CONSTRAINT "organization_id_refs_id_e3e51435" FOREIGN KEY ("organization_id") REFERENCES "sentry_organization" ("id") DEFERRABLE INITIALLY DEFERRED;
CREATE INDEX "sentry_incident_organization_id" ON "sentry_incident" ("organization_id");
ALTER TABLE "sentry_incidentproject"
  ADD CONSTRAINT "incident_id_refs_id_c8831f42" FOREIGN KEY ("incident_id") REFERENCES "sentry_incident" ("id") DEFERRABLE INITIALLY DEFERRED;
CREATE INDEX "sentry_incidentproject_incident_id" ON "sentry_incidentproject" ("incident_id");
```

Comment (mattrobenolt) [reply src/sentry/south_migrations/0476_incidents.py]: Yeah, this is fine, I also don't think we need to set `is_dangerous` flag here unless you're nervous about it.

Comment (wedamija) [reply src/sentry/south_migrations/0476_incidents.py]: Not overly, I'll make it not dangerous.

**Review by markstory — APPROVED** [review]

**Pull request merged**